pep-0696.rst

```diff
@@ -190,38 +196,52 @@ default to the type of ``start`` and step default to ``int | None``.
 
    class slice(Generic[StartT, StopT, StepT]): ...
 
+   reveal_type(slice)  # type is type[slice[StartT = int, StopT = StartT, StepT = int | None]]
    reveal_type(slice())                        # type is slice[int, int, int | None]
    reveal_type(slice[str]())                   # type is slice[str, str, int | None]
    reveal_type(slice[str, bool, timedelta]())  # type is slice[str, bool, timedelta]
 
+   T2 = TypeVar("T2", default=DefaultStrT)
+
+   class Foo(Generic[DefaultStrT, T2]):
+       def __init__(self, a: DefaultStrT, b: T2) -> None: ...
+
+   reveal_type(Foo(1, ""))  # type is Foo[int, str]
+   Foo[int](1, "")          # Invalid: Foo[int, str] cannot be assigned to self: Foo[int, int] in Foo.__init__
+   Foo[int]("", 1)          # Invalid: Foo[str, int] cannot be assigned to self: Foo[int, int] in Foo.__init__
+
 When using a ``TypeVarLike`` as the default to another ``TypeVarLike``.
 Where ``T1`` is the default for ``T2`` the following rules apply.
 
-.. _scoping-rules:
+.. note::
+
+   ``TypeVarTuple``\s as defaults are not supported due to
```

Comment:
This sentence is very hard to understand, could you rewrite it?

Reply:
For starters on this line, you could do

```suggestion
   ``TypeVarTuple``\s are not supported as defaults due to
```

Reply:
It's more that I got lost on the next few lines trying to figure out the structure of the sentence :)

Reply:
Does 
```rst
.. note::

   ``TypeVarTuple``\s are not supported as defaults due to:

   - :ref:`696-scoping-rules` not allowing usage of ``TypeVarLikes``
     from outer scopes
   - Multiple ``TypeVarTuple``\s not being able appear in the same type
     parameter lists to a class as defined in
     :pep:`646#multiple-type-variable-tuples-not-allowed`
   - Function defaults not being supported

   These reasons leave no current valid location for a
   ``TypeVarTuple``, hence, they are not supported.
```
read better?

Reply:
(Fixed a few minor punctuation issues)

Reply:
I would rewrite each bullet point as a sentence, something like
Does 
```rst
.. note::

   ``TypeVarTuple``\s are not supported because:

   - :ref:`696-scoping-rules` does not allow usage of ``TypeVarLikes``
     from outer scopes.
   - Multiple ``TypeVarTuple``\s cannot appear in the type
     parameter list for a single class, as specified in
     :pep:`646#multiple-type-variable-tuples-not-allowed`.
   - ``TypeVar`` defaults in functions are not supported.

   These reasons leave no current valid location where a
   ``TypeVarTuple`` could have a default.
```

Reply:
That reads better now, thanks

Reply:
I just looked at the preview (https://pep-previews--2902.org.readthedocs.build/pep-0696/#using-another-typevarlike-as-default) and the "Note" formatting here looks way too prominent for my taste. Maybe remove the ".. note" and just make this a normal paragraph.

Reply:
Yeah, as the person who suggested that in the first place (because the text was in a manual `Note:` block, especially with the reformat, seems like it makes more sense and flows better as prose now.